### ISSUE TITLE
_open_osfhandle() fails for the program handle returned by ebpfcore.

### ISSUE THREAD

**Issue by saxena-anurag** [microsoft/ebpf-for-windows#287]
_open_osfhandle() is failing currently when called for the program handle returned by ebpfcore. It may have to do with the CreateOptions used in ZwCreateFile, needs to be investigated.

**Comment by dthaler** [comment]
Reopening since there is still the workaround to use sequential ids that needs to be removed if this fixes the underlying issue.

**Comment by Alan-Jowett** [comment]
Assigning back to @saxena-anurag as the driver issue should now be fixed.

**Comment by dthaler** [comment]
@saxena-anurag I can do this one if you prefer

**Comment by Alan-Jowett** [comment]
I am fine doing it, I just couldn't find where the code is that is doing this.

**Comment by dthaler** [comment]
I made a change, I just need to test it

**Comment by saxena-anurag** [comment]
The code has issue#287 mentioned where we need to make the change. Mainly 2 places:

1. Where the fd is generated (function _get_next_file_descriptor())
2. Where we get handle from fd (function _get_handle_from_fd())

I think we will also need to write a platform specific version of _open_osfhandle(), like there is for DeviceIoControl() for UM tests to work.

**Comment by dthaler** [comment]
Yeah, ebpf_handle_user.c just returns integers, not things that _open_osfhandle() will accept
